### PR TITLE
Initialize load testing

### PR DESCRIPTION
**Pull request by yaoxiaoqi**
# Changes

This PR initializes load testing. It starts multiple taskruns and sees if they could
complete in defined time.

It also provides a script to run load testing. Users can use flag `--taskrun-num`
and `--timeout` to specify the amount of taskruns they want to run and the time
in which taskruns should complete.

To run the load testing:

`./test/load-test.sh --taskrun-num 50 --timeout 10m`

/kind feature

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Release notes block has been filled in or deleted (only if no user facing changes)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) to build and release this image.

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

# Release Notes

```release-note
Initialize load testing
```

### CODE REVIEW

File: test/load_test.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+				TaskSpec: &v1beta1.TaskSpec{
+					Steps: []v1beta1.Step{{Container: corev1.Container{
+						Image:   "busybox",
+						Command: []string{"/bin/echo"},
+						Args:    []string{"simple"},
+					}}},
+				},
+			},
+		}
+		if _, err := c.TaskRunClient.Create(ctx, taskrun, metav1.CreateOptions{}); err != nil {
```
Comment (vdemeester) [inline test/load_test.go:67]: Any reason for creating the `TaskRun` synchronously ?

**Comment by bobcatfish** [comment]
Hey @yaoxiaoqi ! You might be interested in the PR I opened around getting some performance measurements up and running: https://github.com/tektoncd/community/pull/277 If you are interested in collaborating on that TEP with me that'd be very cool! In the meantime I'm hesitant to commit scripts to this repo, especially without some kind of automation to ensure they stay up and running.

Basically anything we commit here we should make sure to maintain, so I'm not sure if we want to add this just yet?

**Comment by bobcatfish** [comment]
I think I should probably explain my thought process a bit more!

Basically I think if you (or anyone!) want to run some loadtests and maybe share a report on your findings with the community that is 👍 👍 👍  However I would only commit the code for that to this repo if we want to keep using it over time; if it's a one off for some experimentation it could live in a private repo maybe, or even in experimental.

If we want to start adding some loadtests that are run and maintained by the community, I think we should start with a TEP and gather some feedback before we pursue something; tektoncd/community#277 is specifically about trying to catch performance regressions such as the one we had recently before they happen by starting to define some SLIs and SLOs.

**Comment by yaoxiaoqi** [comment]
> Hey @yaoxiaoqi ! You might be interested in the PR I opened around getting some performance measurements up and running: [tektoncd/community#277](https://github.com/tektoncd/community/pull/277) If you are interested in collaborating on that TEP with me that'd be very cool! In the meantime I'm hesitant to commit scripts to this repo, especially without some kind of automation to ensure they stay up and running.
> 
> Basically anything we commit here we should make sure to maintain, so I'm not sure if we want to add this just yet?

Hi @bobcatfish ! I'm glad to collaborate on the TEP with you. And I think it's actually kind of unconsidered to add this script now. It would be better if I implemented a maintainable load tests/ performance tests after the tep had been discussed thoroughly. Anyway, this pr meant to be a load testing sample and could be hold if this is not the time. Thanks for your suggestions!

**Comment by bobcatfish** [comment]
sounds good @yaoxiaoqi !! in that case maybe we can close this PR for now? I'll try to get a doc started that we can collaborate in ASAP :D

Also very interested to see the results of the tests you've run so far :D

**Comment by yaoxiaoqi** [comment]
Sounds great @bobcatfish ! I will close this PR. Please send me the doc when you're ready.

Using this script I found that tekton pipelines could run 500 simple taskruns in ~800 seconds on a 12-cpu 45G-memory gke cluster. I will try some cpu bound or i/o bound taskruns and pipelineruns in the future to see the pipelines performance in different situations.

**Pull request closed without merging**